commit aedbcb73266c708401ff4185d0ca63b7cf5fd894
Author: Vincent Batts <vbatts@hashbangbash.com>
Date:   Tue May 1 11:33:38 2018 -0400

    server/hot_restart_impl: gcc8 errors for memset
    
    ```
    source/server/hot_restart_impl.cc: In member function 'virtual void Envoy::Server::HotRestartImpl::free(Envoy::Stats::RawStatData&)':
    source/server/hot_restart_impl.cc:171:46: error: 'void* memset(void*, int, size_t)' clearing an object of type 'struct Envoy::Stats::RawStatData' with no trivial copy-assignment [-Werror=class-memaccess]
       memset(&data, 0, Stats::RawStatData::size());
    ```

diff --git a/source/server/hot_restart_impl.cc b/source/server/hot_restart_impl.cc
index 6e81f7de1..97b301c9f 100644
--- a/source/server/hot_restart_impl.cc
+++ b/source/server/hot_restart_impl.cc
@@ -168,7 +168,7 @@ void HotRestartImpl::free(Stats::RawStatData& data) {
   }
   bool key_removed = stats_set_->remove(data.key());
   ASSERT(key_removed);
-  memset(&data, 0, Stats::RawStatData::size());
+  memset(static_cast<void *>(&data), 0, Stats::RawStatData::size());
 }
 
 int HotRestartImpl::bindDomainSocket(uint64_t id) {
